### PR TITLE
2D Boolean Set Operations: Improve Dcoumentation of Free Function

### LINKED ISSUE #4998: [Boolean Operations] Assertion fail when intersecting general polygons with polyline traits
## Issue Details

In some cases, using `Arr_polyline_traits_2` with boolean operations leads to assertion failures. One simple example is the one given hereafter, where we intersect a square with itself. The resulting `General_polygon_set_2` is detected as invalid:

```
CGAL error: assertion violation!
Expression : is_valid()
File       : /home/simon/programs/cgal/master/Boolean_set_operations_2/include/CGAL/Boolean_set_operations_2/Gps_on_surface_base_2.h
Line       : 1489
```
If I dig a bit more in the validation code, it is specifically this part (`Gps_on_surface_base_2.h` line 576) that fails:
```c++
      const X_monotone_curve_2&  cv = he->curve();
      const bool                 is_cont = he->face()->contained();
      const Comparison_result    he_res =
        ((Arr_halfedge_direction)he->direction() == ARR_LEFT_TO_RIGHT) ?
        SMALLER : LARGER;
      const bool                 has_same_dir = (cmp_endpoints(cv) == he_res);

      if ((is_cont && !has_same_dir) || (!is_cont && has_same_dir)) {
        return false;
      }
```

The problem seems to come from an inversion of orientation of an edge. If I define `CGAL_ALWAYS_LEFT_TO_RIGHT`, then I run into an assertion fail much earlier:

```
CGAL warning: check violation!
Expression : closed
File       : /home/simon/programs/cgal/master/Boolean_set_operations_2/include/CGAL/Boolean_set_operations_2/Gps_polygon_validation.h
Line       : 296
Explanation: The polygon's boundary is not closed.
```

## Source Code

```c++
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Boolean_set_operations_2.h>
#include <CGAL/Arr_polyline_traits_2.h>
#include <CGAL/General_polygon_2.h>

#include <boost/function_output_iterator.hpp>

using Kernel = CGAL::Exact_predicates_exact_constructions_kernel;
using Point_2 = Kernel::Point_2;

using Segment_traits = CGAL::Arr_segment_traits_2<Kernel>;
using Segment_curve = Segment_traits::X_monotone_curve_2;

using Polyline_traits = CGAL::Arr_polyline_traits_2<Segment_traits>;
using Polyline = Polyline_traits::Curve_2;
using Polyline_curve = Polyline_traits::X_monotone_curve_2;

using Polygon = CGAL::General_polygon_2<Polyline_traits>;

int main()
{
  // Create square polyline
  Point_2 p00 (0, 0), p01 (0, 1), p10 (1, 0), p11 (1, 1);
  std::array<Segment_curve, 4> segment_curves
    = { Segment_curve(p00, p10),
        Segment_curve(p10, p11),
        Segment_curve(p11, p01),
        Segment_curve(p01, p00) };
  Polyline polyline (segment_curves.begin(), segment_curves.end());

  // Create Polyline x monotone curves
  std::vector<Polyline_curve> polyline_curves;
  Polyline_traits traits;
  Polyline_traits::Make_x_monotone_2 make_x_monotone_2
    = traits.make_x_monotone_2_object();
  make_x_monotone_2
    (polyline,
     boost::make_function_output_iterator
     ([&](const CGAL::Object& obj)
      {
        // All objects should be curves
        const Polyline_curve* xc
          = CGAL::object_cast<Polyline_curve>(&obj);
        CGAL_assertion (xc);
        polyline_curves.push_back (*xc);
      }));

  // Intersect polygon with itself
  Polygon pa (polyline_curves.begin(), polyline_curves.end());
  Polygon pb (polyline_curves.begin(), polyline_curves.end());
  CGAL::intersection(pa, pb, CGAL::Emptyset_iterator());

  return EXIT_SUCCESS;
}
```

### PR DESCRIPTION
**Pull request by efifogel**
## Summary of Changes

1. Added missing references to `oriented_side(Point_2, ....)`
2. Fixed references to all BSO_2 functions.
3. Fixed a bug in the `Arr_polycurve_traits_2::Intersect_2` functor. (This is unrelated to items (1) and (2) above, but it's only a one-liner.

It turns out that the `oriented_side()` overloaded functions that accept a point and a polygon were not documented.
This PR adds the missing functions. 
White at it, I've fixed the documentation of all other functions.
For example, we typically do not put too much emphasize of documenting a traits optional parameter.
I've decided to do it after all, because there are related preconditions that the user should be informed bout when passing a traits.

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #4998 and more
* Feature/Small Feature (if any): [`Features/Small_Features/Improve_Dcoumentation_of_2D_Boolean_Set_Operations_Free_Functions`](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Improve_Dcoumentation_of_2D_Boolean_Set_Operations_Free_Functions)
* Pre-approved by @sloriot: 27 January 2021
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

**Comment by lrineau** [comment]
This must be submitted, and then reviewed as a small feature.

**Comment by efifogel** [comment]
ok.
BW, the small fix that I've added on the way can wait as well, unless Simon
wants it in. In this case it can be cherry-picked.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
build:test-build-doc

**Comment by lrineau** [comment]
build:test-build-doc

**Comment by lrineau** [comment]
build:test-build-doc3

**Comment by lrineau** [comment]
build:test-build-doc4

**Comment by lrineau** [comment]
> build:test-build-doc4

@efifogel For information, we (Maxime and I) were trying to test a new feature that we are setting up: the automatic building of the documentation by Github itself. ...That does not work. We will stop the tests until Maxime has finished more testing in another repository. Sorry for the spam on your PR.

**Comment by efifogel** [comment]
No problem
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
build:test-build-doc-5

**Comment by lrineau** [comment]
build:test-build-doc-6

**Comment by lrineau** [comment]
build:test-build-doc-7

**Comment by lrineau** [comment]
build:test-build-doc-8

**Comment by lrineau** [comment]
build:test-build-doc-10

**Comment by lrineau** [comment]
build:test-build-doc-9

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 1]
```diff
@@ -1,833 +1,2166 @@
...
 
-/*!
-\addtogroup boolean_complement Complement Functions
-\ingroup PkgBooleanSetOperations2Ref
-\anchor ref_bso_complement
+/*! \addtogroup boolean_complement Complement Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_complement
+ *
+ * There are several overloaded function templates called `complement()` that
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:7]:
```suggestion
 * There are several overloaded template functions called `complement()` that
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 2]
```diff
@@ -1,833 +1,2166 @@
...
-the `complement function` writes them into an output iterator
-`oi`.
+/// @{
 
-\param pgn The input polygon for the `complement` function. It may be of the type
-`Polygon_2`, `General_polygon_2`, `Polygon_with_holes_2`, or
-`General_polygon_with_holes_2`.
+//////// Traits-less
 
+/*! Computes the complement of a polygon.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:44]:
```suggestion
/*! computes the complement of a polygon.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 3]
```diff
@@ -1,833 +1,2166 @@
...
+ * \param res the complement of \p pgn.
+ * \pre `%ArrTraits` must be a model of the concept
+ *      `ArrangementDirectionalXMonotoneTraits_2`.
+ */
+template <typename ArrTraits>
+void
+complement(const General_polygon_2<ArrTraits>& pgn,
+           General_polygon_with_holes_2<General_polygon_2<Arr_traits> >& res);
+
+/*! Computes the complement of a polygon with holes.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:63]:
```suggestion
/*! computes the complement of a polygon with holes.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 4]
```diff
@@ -1,833 +1,2166 @@
...
 
+/*! Computes the complement of a polygon.
+ * \param pgn the input polygon.
+ * \param res the resulting complement of \p pgn.
+ */
+template <typename Kernel, typename Container>
+void complement(const Polygon_2<Kernel, Container>& pgn,
+                Polygon_with_holes_2<Kernel, Container>& res);
+
+/*! Computes the complement of a general polygon.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:52]:
```suggestion
/*! computes the complement of a general polygon.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 5]
```diff
@@ -1,833 +1,2166 @@
...
+ * \param oi the output iterator for the result.
+ *           Its dereference type must be convertible to
+ *             `Polygon_with_holes_2<Kernel, Container>`.
+ * \return the past-the-end iterator of the output container.
+ */
+template <typename Kernel, typename Container, typename OutputIterator>
+OutputIterator complement(const Polygon_with_holes_2<Kernel, Container>& pgn,
+                          OutputIterator oi);
 
+/*! Computes the complement of a general polygon with holes.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:74]:
```suggestion
/*! computes the complement of a general polygon with holes.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 6]
```diff
@@ -1,833 +1,2166 @@
...
-\sa \link boolean_join `CGAL::join()` \endlink
-\sa \link boolean_difference `CGAL::difference()` \endlink
-\sa \link boolean_symmetric_difference `CGAL::symmetric_difference()` \endlink
-*/
-/// @{
+//////// With Traits
 
-/*!
- writes the complement of the polygon `pgn` into the  polygon with holes `res`.
+/*! Computes the complement of a polygon.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:90]:
```suggestion
/*! computes the complement of a polygon.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 7]
```diff
@@ -1,833 +1,2166 @@
...
-void complement(const Polygon_2<Kernel, Container> & pgn, Polygon_with_holes_2<Kernel, Container> & res);
-
-/*!
-  writes the complement of the general polygon `pgn` into the  general polygon with holes `res`.
+template <typename Kernel, typename Container, typename GpsTraits>
+void complement(const Polygon_2<Kernel, Container>& pgn,
+                Polygon_with_holes_2<Kernel, Container>& res,
+                const GpsTraits& traits);
+
+/*! Computes the complement of a general polygon.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:101]:
```suggestion
/*! computes the complement of a general polygon.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 8]
```diff
@@ -1,833 +1,2166 @@
...
-/*!
-  writes the complement of the polygon with holes `pgn` into the  output iterator `oi`.
-  The value type of `oi` is `Polygon_with_holes_2`.
+template <typename ArrTraits, typename GpsTraits>
+void
+complement(const General_polygon_2<ArrTraits>& pgn,
+           General_polygon_with_holes_2<General_polygon_2<Arr_traits> >& res,
+           const GpsTraits& traits);
+
+/*! Computes the complement of a polygon with holes.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:115]:
```suggestion
/*! computes the complement of a polygon with holes.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 9]
```diff
@@ -1,833 +1,2166 @@
...
-/*!
-  writes the complement of the general polygon with holes `pgn` into the  output iterator `oi`.
-  The value type of `oi` is `General_polygon_with_holes_2`.
+template <typename Kernel, typename Container, typename OutputIterator,
+          typename GpsTraits>
+OutputIterator complement(const Polygon_with_holes_2<Kernel, Container>& pgn,
+                          OutputIterator oi,
+                          const GpsTraits& traits);
+
+/*! Computes the complement of the general polygon with holes.
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:130]:
```suggestion
/*! computes the complement of the general polygon with holes.
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 10]
```diff
@@ -1,833 +1,2166 @@
...
-
-\sa \link boolean_do_intersect `CGAL::do_intersect()` \endlink
-\sa \link boolean_intersection `CGAL::intersection()` \endlink
-\sa \link boolean_join `CGAL::join()` \endlink
-\sa \link boolean_symmetric_difference `CGAL::symmetric_difference()` \endlink
+/*! \addtogroup boolean_difference Difference Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_difference
+ *
+ * There are several overloaded function templates called `difference()` that
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:156]:
```suggestion
 * There are several overloaded template functions called `difference()` that
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 11]
```diff
@@ -1,833 +1,2166 @@
...
-\sa \link boolean_join `CGAL::join()` \endlink
-\sa \link boolean_difference `CGAL::difference()` \endlink
-\sa \link boolean_symmetric_difference `CGAL::symmetric_difference()` \endlink
-
-*/
+/*! \addtogroup boolean_do_intersect Intersection Testing Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_do_intersect
+ *
+ * There are several overloaded function templates called `do_intersect()`
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:476]:
```suggestion
 * There are several overloaded template functions called `do_intersect()`
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 12]
```diff
@@ -1,833 +1,2166 @@
...
+template <typename Polygon>
+bool do_intersect(const General_polygon_with_holes_2<Polygon>& pgn1,
+                  const General_polygon_with_holes_2<Polygon>& pgn2);
+
+/*! Given a range of polygons or a range of polygons with holes (resp. a range
+ * of general polygons or a range of general polygons with holes) determines
+ * whether the open polygons (resp. general polygons) in the range have a common
+ * point.
+ * \param begin the first iterator of the input range. It's value type is
+ *        either Polygon_2 (resp. General_polygon_2) or Polygon_with_holes
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:608]:
```suggestion
 *        either `Polygon_2` (resp. `General_polygon_2`) or `Polygon_with_holes`
```

Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:608]:
```suggestion
 *        either `Polygon_2` (resp. `General_polygon_2`) or `Polygon_with_holes`
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 13]
```diff
@@ -1,833 +1,2166 @@
...
+ *        (resp. General_polygon_with_holes_2).
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:609]:
```suggestion
 *        (resp. `General_polygon_with_holes_2`).
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 14]
```diff
@@ -1,833 +1,2166 @@
...
+ * \param end the past-the-end iterator of the input range. It's value type is
+ *        either Polygon_2 (resp. General_polygon_2) or Polygon_with_holes
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:611]:
```suggestion
 *        either `Polygon_2` (resp. `General_polygon_2`) or `Polygon_with_holes`
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 15]
```diff
@@ -1,833 +1,2166 @@
...
+ *        (resp. General_polygon_with_holes_2).
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:612]:
```suggestion
 *        (resp. `General_polygon_with_holes_2`).
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 16]
```diff
@@ -1,833 +1,2166 @@
...
-\sa \link boolean_difference `CGAL::difference()` \endlink
-\sa \link boolean_symmetric_difference `CGAL::symmetric_difference()` \endlink
-
-*/
-
+/*! \addtogroup boolean_intersection Intersection Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_intersection
+ *
+ * There are several overloaded function templates called `intersection()` that
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:810]:
```suggestion
 * There are several overloaded template functions called `intersection()` that
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 17]
```diff
@@ -1,833 +1,2166 @@
...
-\sa \link boolean_intersection `CGAL::intersection()` \endlink
-\sa \link boolean_difference `CGAL::difference()` \endlink
-\sa \link boolean_symmetric_difference `CGAL::symmetric_difference()` \endlink
-
-*/
+/*! \addtogroup boolean_join Union Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_union
+ *
+ * There are several overloaded function templates called `join()` that
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:1027]:
```suggestion
 * There are several overloaded template functions called `join()` that
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 18]
```diff
@@ -1,833 +1,2166 @@
...
-
-\sa \link boolean_do_intersect `CGAL::do_intersect()` \endlink
-
-*/
-
+/*! \addtogroup boolean_oriented_side Oriented Side Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_oriented_side
+ *
+ * There are several overloaded function templatess called `Oriented_side()`
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:1397]:
```suggestion
 * There are several overloaded template functions called `Oriented_side()`
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 19]
```diff
@@ -1,833 +1,2166 @@
...
-\sa \link boolean_intersection `CGAL::intersection()` \endlink
-\sa \link boolean_join `CGAL::join()` \endlink
-\sa \link boolean_difference `CGAL::difference()` \endlink
-
-*/
+/*! \addtogroup boolean_symmetric_difference Symmetric Difference Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_symmetric_difference
+ *
+ * There are several overloaded function templates called
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:1728]:
```suggestion
 * There are several overloaded template functions called
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h [hunk 20]
```diff
@@ -1,833 +1,2166 @@
...
-
-*/
-
+/*! \addtogroup boolean_oriented_side Oriented Side Functions
+ * \ingroup PkgBooleanSetOperations2Ref
+ * \anchor ref_bso_oriented_side
+ *
+ * There are several overloaded function templatess called `Oriented_side()`
+ * that compute the relative position of either (i) two polygons or (ii) a
+ * point and a polygon. This group of function templates is divided into two
```
Comment (sloriot) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:1399]:
```suggestion
 * point and a polygon. This group of template functions is divided into two
```

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h (hunk 15 above)
Comment (sloriot) [reply Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:612]: There are many similar missing backticks, I won't mark them all in the review...

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h (hunk 1 above)
Comment (efifogel) [reply Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h:7]: Nop, it should be "function template"; see, e.g., https://en.cppreference.com/w/cpp/language/function_template

**Comment by efifogel** [comment]
Thanks Sebastien for the review, suggested fixes, and triggering the
missing links.
I went through it again and fixed all (the missing links).
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by efifogel** [comment]
I've cherry-picked the small fix into the branch cgal-public-dev/Arrangement-Fix_polycurve_intersection-efif-GF, which will be merged as part of PR #5091 (Fix polycurve overlap intersection)

**Comment by lrineau** [comment]
TODO: resolve the conflicts, and make sure the modifications of `Installtion/CHANGES.md` are in the section about 5.3, and not 5.2.

@sloriot You have already reviewed this PR. Is it in a state you can pre-approve the small feature?

**Comment by lrineau** [comment]
I have re-added the labels:
  - TODO, for https://github.com/CGAL/cgal/pull/5050#issuecomment-767474032,
  - "not yet approved", because the small feature is only pre-approved.

@efifogel: Maxime will start testing the pull-request as soon as the conflicts are resolved (and the `CHANGES.md` modified to target 5.3 instead of 5.2). And the pull-request can be merged into `master` into two weeks, if the tests are good, once the delay of two weeks after the pre-approval is over.

**Comment by efifogel** [comment]
Thanks.
This is not a problem.
I'm simply waiting for this one before I finish "Boolean Operations: Use
polyline-based GPS (#5094)"
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
> Thanks. This is not a problem. I'm simply waiting for this one before I finish "Boolean Operations: Use polyline-based GPS (#5094)"

@efifogel  Then you have to merge `master` in it, resolve the conflicts, and reassign the paragraph in `CHANGES.md` to the section about CGAL-5.3.

**Comment by efifogel** [comment]
Right; all done.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-65.shtml

**Pull request merged**